gfx/common/x11_common.c

```diff
@@ -306,9 +306,17 @@ static void x11_suspend_screensaver_xdg_screensaver(Window wnd, bool enable)
 
 void x11_suspend_screensaver(Window wnd, bool enable)
 {
-    x11_suspend_screensaver_xdg_screensaver(wnd, enable);
 #ifdef HAVE_DBUS
     x11_suspend_screensaver_dbus(enable);
+    if (dbus_screensaver_cookie == 0)
+    {
+        // Only attempt xdg-screensaver method if dbus didn't succeed
+#endif
+
+        x11_suspend_screensaver_xdg_screensaver(wnd, enable);
+
+#ifdef HAVE_DBUS
+    }
```

Comment:
Hm, I had a look at having `x11_suspend_screensaver_dbus()` return a boolean it if succeeded or not but it's actually a bit tricky because it also depends on `enabled`. If `enabled` is false, then `x11_suspend_screensaver_dbus()` actually un-suspends it, so returning true/false to indicate inhibition success doesn't make sense anymore. It also depends if the screensaver was already suspended or not which muddies the meaning of "did I succeed to inhibit or not".

I agree the design of the screensaver inhibition is a bit off. I think it could be simplified by having the calling code directly call "inhibit" or "resume" only if the suspend menu option is enabled, and not call anything if it's not the case, rather than passing the `enabled` boolean value down... However that's to be fixed separately.

> Add a || enable and I'll accept it

I'm not sure why that's required, since `enabled` is passed down to `x11_suspend_screensaver_dbus()` (and `x11_suspend_screensaver_xdg_screensaver()`) which deals with it appropriately depending on the situation (screensaver is already inhibited, etc)? If we just guard it with `|| enabled` then it may never be un-inhibited?